flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -188,7 +190,7 @@ public int getNumberOfSubpartitions() {
 		return subpartitions.length;
 	}
 
-	@Override
+	@VisibleForTesting
 	public BufferProvider getBufferProvider() {
```

Comment:
can this be package private?

Reply:
we can actually remove it and use `getBufferPool` in `LocalInputChannelTest` where it is used.

Reply:
It could not be package private, but we could remove it to use `getBufferPool` instead. I could submit a fixup for it.